src/collection.js

```diff
@@ -525,12 +525,13 @@ export default class Collection {
       updated.last_modified = oldRecord.last_modified;
     }
     // If only local fields have changed, then keep record as synced.
+    // If status is created, keep record as created.
+    // If status is deleted, mark as updated.
     const isIdentical = oldRecord && recordsEqual(oldRecord, updated, this.localFields);
     const keepSynced = isIdentical && oldRecord._status == "synced";
-    let newStatus = (keepSynced || synced) ? "synced" : "updated";
-    if (!oldRecord) {
-      newStatus = "created";
-    }
+    const neverSynced = !oldRecord || (oldRecord && oldRecord._status == "created");
+    const newStatus = (keepSynced || synced) ? "synced"
+                                             : neverSynced ? "created" : "updated";
```

Comment:
I still feel like this is too complicated or ugly to live. (Ternary operators inside ternary operators!!) How about this?

```javascript
let oldStatus = oldRecord && oldRecord._status;
let newStatus;
if (oldStatus == "synced" && !isIdentical) {
    // If the old record was "synced" but we've changed it, the new record should be "updated", 
    // even if the user didn't update the status themselves
    newStatus = "updated";
} else if (oldStatus == "deleted") {
    // If the old record was "deleted" (i.e. deleted locally but not synced) and 
    // we're doing a save on it, we should "undelete" it. From the server's 
    // perspective, it's "updated"
    newStatus = "updated";
} else if (synced) {
    // The `synced` option means we should force it to "synced" status
    newStatus = "synced";
} else {
    // Otherwise, default to whatever the user gave us. 
    // If that's missing, copy from the old record. 
    // If there was no old record, then we're "created".
    newStatus = updated._status || oldStatus || "created";
}
```

It's a lot longer, but I feel like it's a lot simpler by virtue of only specifying the cases when a status should change instead of the cases where it shouldn't.